src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/PropertyBuilderImpl.cs

```diff
@@ -33,7 +33,7 @@ internal PropertyBuilderImpl(string name, PropertyAttributes attributes, Calling
             _name = name;
             _attributes = attributes;
             _callingConvention = callingConvention;
-            _propertyType = returnType;
+            _propertyType = returnType ?? containingType.GetModuleBuilder().GetTypeFromCoreAssembly(CoreTypeId.Void);
```

Comment:
Existing `AssemblyBuilder` adds void type for null return type when populating the signature, now this scenario causing NRE in the SignatureHelper, I could fix it in the signature only, but to avoid NRE's for other case setting the return type to void here.

Reply:
The constructor parameter, `Type returnType` isn't marked as nullable. If a `null` value is sneaking through, we should fix that or mark the parameter as nullable.

Reply:
Sure, that makes sense. I don't think we should throw either, but I do like to have the nullable indicators correct so we can check these things.

Reply:
> Existing AssemblyBuilder adds void type for null return type when populating the signature,

I assume this behavior exists in previous versions as well?

Reply:
Yes, `RuntimePropertyBuilder` constructor was allowing null, but not handling null which could cause NRE further if `SetConstant` called for the property, the null has been resolved only for `GetPropertySigHelper`. I see same behavior in .NET framework.

Now with this PR the `null` return type will be handled to void type in the `RuntimePropertyBuilder` constructor, same as in `GetPropertySigHelper`.